From c827e811489b924ccab8821db0ada7b06a222f9f Mon Sep 17 00:00:00 2001
From: Sandeep Patil <sspatil@google.com>
Date: Mon, 9 Sep 2019 08:34:43 -0700
Subject: [PATCH 257/830] ANDROID: staging: ion: uapi: match the existing heap
 type enums
X-NVConfidentiality: public

Change the enup order for ion heap types to match how it was before so
as to not break any userspace code that depends on them.

Bug: 140507100
Test: ion-unit-tests

Change-Id: Id68c50a9bd3eda2ce46bc88a326a120c52d1110f
Signed-off-by: Sandeep Patil <sspatil@google.com>
---
 include/uapi/linux/ion.h | 10 +++++-----
 1 file changed, 5 insertions(+), 5 deletions(-)

diff --git a/include/uapi/linux/ion.h b/include/uapi/linux/ion.h
index 09dbbfaf3718..ad773e5fa4d2 100644
--- a/include/uapi/linux/ion.h
+++ b/include/uapi/linux/ion.h
@@ -31,8 +31,8 @@
 enum ion_heap_type {
 	ION_HEAP_TYPE_SYSTEM = 0,
 	ION_HEAP_TYPE_SYSTEM_CONTIG = 1,
-	ION_HEAP_TYPE_CHUNK = 2,
-	ION_HEAP_TYPE_CARVEOUT = 3,
+	ION_HEAP_TYPE_CARVEOUT = 2,
+	ION_HEAP_TYPE_CHUNK = 3,
 	ION_HEAP_TYPE_DMA = 4,
 	/* reserved range for future standard heap types */
 	ION_HEAP_TYPE_CUSTOM = 16,
@@ -61,10 +61,10 @@ enum ion_heap_type {
 enum ion_heap_id {
 	ION_HEAP_SYSTEM = (1 << ION_HEAP_TYPE_SYSTEM),
 	ION_HEAP_SYSTEM_CONTIG = (ION_HEAP_SYSTEM << 1),
-	ION_HEAP_CHUNK = (ION_HEAP_SYSTEM_CONTIG << 1),
-	ION_HEAP_CARVEOUT_START = (ION_HEAP_CHUNK << 1),
+	ION_HEAP_CARVEOUT_START = (ION_HEAP_SYSTEM_CONTIG << 1),
 	ION_HEAP_CARVEOUT_END = (ION_HEAP_CARVEOUT_START << 4),
-	ION_HEAP_DMA_START = (ION_HEAP_CARVEOUT_END << 1),
+	ION_HEAP_CHUNK = (ION_HEAP_CARVEOUT_END << 1),
+	ION_HEAP_DMA_START = (ION_HEAP_CHUNK << 1),
 	ION_HEAP_DMA_END = (ION_HEAP_DMA_START << 7),
 	ION_HEAP_CUSTOM_START = (ION_HEAP_DMA_END << 1),
 	ION_HEAP_CUSTOM_END = (ION_HEAP_CUSTOM_START << 15),
-- 
2.17.1

